packages/xrpl/test/integration/utils.ts

```diff
@@ -1,45 +1,133 @@
 import { assert } from 'chai'
-import _ from 'lodash'
+import omit from 'lodash/omit'
 import { decode } from 'ripple-binary-codec'
-import { Client, Wallet, AccountInfoRequest } from 'xrpl-local'
+import {
+  Client,
+  Wallet,
+  AccountInfoRequest,
+  type SubmitResponse,
+  TimeoutError,
+  NotConnectedError,
+} from 'xrpl-local'
 import { Payment, Transaction } from 'xrpl-local/models/transactions'
 import { hashSignedTx } from 'xrpl-local/utils/hashes'
 
 const masterAccount = 'rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh'
 const masterSecret = 'snoPBrXtMeMyMHUVTgbuqAfg1SUTb'
 
-export async function ledgerAccept(client: Client): Promise<void> {
+// let ledgerAcceptMutex = false
+
+export async function ledgerAccept(
+  client: Client,
+  retries?: number,
+): Promise<void> {
   const request = { command: 'ledger_accept' }
-  await client.connection.request(request)
+
+  try {
+    await client.connection.request(request)
+  } catch (error) {
+    if (retries === undefined) {
+      setTimeout(() => {
+        ledgerAccept(client, 10)
+      }, 1000)
+    } else if (retries > 0) {
+      setTimeout(() => {
+        ledgerAccept(client, retries - 1)
+      }, 1000)
+    } else {
+      throw error
+    }
+  }
 }
 
-export function subscribeDone(client: Client, done: Mocha.Done): void {
+export function subscribeDone(client: Client): void {
   client.removeAllListeners()
-  done()
+}
+
+async function runCommand({
+  client,
+  transaction,
+  wallet,
+  retry = { count: 5, delayMs: 1000 },
+}: {
+  client: Client
+  transaction: Transaction
+  wallet: Wallet
+  retry?: {
+    count: number
+    delayMs: number
+  }
+}): Promise<SubmitResponse> {
+  let response: SubmitResponse
+  try {
+    response = await client.submit(transaction, { wallet })
+
+    // Retry if another transaction finished before this one
+    while (
+      ['tefPAST_SEQ', 'tefMAX_LEDGER'].includes(
+        response.result.engine_result,
+      ) &&
+      retry.count > 0
+    ) {
+      // eslint-disable-next-line no-param-reassign -- we want to decrement the count
+      retry.count -= 1
+      // eslint-disable-next-line no-await-in-loop, no-promise-executor-return -- We are waiting on retries
+      await new Promise((resolve) => setTimeout(resolve, retry.delayMs))
+      // eslint-disable-next-line no-await-in-loop -- We are retryhing in a loop on purpose
+      response = await client.submit(transaction, { wallet })
+    }
+  } catch (error) {
+    if (error instanceof TimeoutError || error instanceof NotConnectedError) {
+      // retry
+      return runCommand({
+        client,
+        transaction,
+        wallet,
+        retry: {
+          ...retry,
+          count: retry.count > 0 ? retry.count - 1 : 0,
```

Comment:
Does this accidentally double decrement since we did a param re-assign above on line 73 with `retry.count -= 1`?

Reply:
No, because this handles the case where the request throws an error versus completing successfully but not with the successful code we want. The loop above does a retry during the happy path while this catch block uses recursion to retry in the event of an unexpected exception.